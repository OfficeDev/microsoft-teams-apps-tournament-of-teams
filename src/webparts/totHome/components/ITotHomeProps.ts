import { WebPartContext } from "@microsoft/sp-webpart-base";
export interface ITotHomeProps {
  description: string;
  context: WebPartContext;
  siteUrl: string;
}
