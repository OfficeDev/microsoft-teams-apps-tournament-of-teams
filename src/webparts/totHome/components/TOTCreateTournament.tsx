import * as React from "react";
import { WebPartContext } from "@microsoft/sp-webpart-base";
import commonServices from "../common/CommonServices";
import * as stringsConstants from "../constants/strings";
import styles from "../scss/TOTCreateTournament.module.scss";

//React Boot Strap
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";

//FluentUI controls
import { TextField } from "@fluentui/react/lib/TextField";
import { IButtonStyles, PrimaryButton } from "@fluentui/react";
import { Label } from "@fluentui/react/lib/Label";
import { Icon, IIconProps } from '@fluentui/react/lib/Icon';
import { mergeStyleSets } from '@fluentui/react/lib/Styling';
import { TooltipHost, ITooltipHostStyles } from '@fluentui/react/lib/Tooltip';

//PNP
import {
  TreeView,
  ITreeItem,
  TreeViewSelectionMode,
} from "@pnp/spfx-controls-react/lib/TreeView";
import { ITextFieldStyles } from "office-ui-fabric-react/lib/components/TextField/TextField.types";
import * as LocaleStrings from 'TotHomeWebPartStrings';

export interface ICreateTournamentProps {
  context?: WebPartContext;
  siteUrl: string;
  onClickCancel: Function;
}

interface ICreateTournamentState {
  actionsList: ITreeItem[];
  tournamentName: string;
  tournamentDescription: string;
  selectedActionsList: ITreeItem[];
  tournamentError: Boolean;
  actionsError: Boolean;
  showForm: Boolean;
  showSuccess: Boolean;
  showError: Boolean;
  errorMessage: string;
}

const calloutProps = { gapSpace: 0 };

const hostStyles: Partial<ITooltipHostStyles> = { root: { display: 'inline-block', cursor: 'pointer' } };

const classes = mergeStyleSets({
  icon: {
    fontSize: '16px',
    paddingLeft: '10px',
    paddingTop: '6px',
    fontWeight: 'bolder',
    color: '#1d0f62'
  }
});

const labelStyles: Partial<ITextFieldStyles> = {
  subComponentStyles: {
    label: {
      root: {
        textAlign: "left",
        font: "normal normal 600 18px/24px Segoe UI",
        letterSpacing: "0px",
        color: "#000000",
        opacity: 1
      }
    }
  }
};

const backBtnStyles: Partial<IButtonStyles> = {
  root: {
    borderColor: "#33344A",
    backgroundColor: "white",
  },
  rootHovered: {
    borderColor: "#33344A",
    backgroundColor: "white",
    color: "#000003"
  },
  rootPressed: {
    borderColor: "#33344A",
    backgroundColor: "white",
    color: "#000003"
  },
  icon: {
    fontSize: "17px",
    fontWeight: "bolder",
    color: "#000003",
    opacity: 1
  },
  label: {
    font: "normal normal bold 14px/24px Segoe UI",
    letterSpacing: "0px",
    color: "#000003",
    opacity: 1,
    marginTop: "-3px"
  }
};

const addIcon: IIconProps = { iconName: 'Add' };
const backIcon: IIconProps = { iconName: 'NavigateBack' };

export default class TOTCreateTournament extends React.Component<
  ICreateTournamentProps,
  ICreateTournamentState
> {
  public createTrmtTreeViewRef: React.RefObject<HTMLDivElement>;
  constructor(props: ICreateTournamentProps, state: ICreateTournamentState) {
    super(props);
    //Set default values for state
    this.createTrmtTreeViewRef = React.createRef();
    this.state = {
      actionsList: [],
      tournamentName: "",
      tournamentDescription: "",
      selectedActionsList: [],
      tournamentError: false,
      actionsError: false,
      showForm: true,
      showSuccess: false,
      showError: false,
      errorMessage: "",
    };

    //Bind Methods
    this.getActions = this.getActions.bind(this);
    this.handleInput = this.handleInput.bind(this);
    this.onActionSelected = this.onActionSelected.bind(this);
    this.saveTournament = this.saveTournament.bind(this);
  }

  //Get Actions from Master list and bind it to treeview on app load
  public componentDidMount() {
    //Get Actions from Master list and bind it to Treeview
    this.getActions();
  }

  public componentDidUpdate(prevProps: Readonly<ICreateTournamentProps>, prevState: Readonly<ICreateTournamentState>, snapshot?: any): void {

    //Update aria-label attribute to all Create Tournament Treeview's Checkbox inputs
    if (prevState.actionsList.length !== this.state.actionsList.length) {
      const checkboxes = this.createTrmtTreeViewRef.current?.getElementsByTagName('input');
      for (let i = 0; i < checkboxes.length; i++) {
        checkboxes[i].setAttribute("aria-label", checkboxes[i].getAttribute('id'));
      }
    }
  }

  //Get Actions from Master list and bind it to Treeview
  private async getActions() {
    console.log(stringsConstants.TotLog + "Getting actions from master list.");
    try {
      //Get all actions from 'Actions List'  to bind it to Treeview
      let commonServiceManager: commonServices = new commonServices(
        this.props.context,
        this.props.siteUrl
      );
      const allActionsArray: any[] = await commonServiceManager.getAllListItems(
        stringsConstants.ActionsMasterList
      );
      var treeItemsArray: ITreeItem[] = [];

      //Loop through all actions and build parent nodes(Categories) for Treeview
      await allActionsArray.forEach((vAction) => {
        const tree: ITreeItem = {
          key: vAction["Category"],
          label: vAction["Category"],
          children: [],
        };
        //Check if Category is already added to the Treeview. If yes, skip adding.
        var found = treeItemsArray.some((value) => {
          return value.label === vAction["Category"];
        });

        //Add category to Treeview only if it doesnt exists already.
        if (!found) treeItemsArray.push(tree);
      });

      //Loop through all actions and build child nodes(Actions) to the Treeview
      await allActionsArray.forEach((vAction) => {
        const tree: ITreeItem = {
          key: vAction.Id,
          label: vAction["Title"],
          data:
            vAction["Category"] +
            stringsConstants.StringSeperator +
            vAction["HelpURL"],
          subLabel:
            vAction["Points"] +
            stringsConstants.PointsDisplayString +
            vAction["Description"],
        };
        var treeCol: Array<ITreeItem> = treeItemsArray.filter((value) => {
          return value.label == vAction["Category"];
        });
        if (treeCol.length != 0) {
          treeCol[0].children.push(tree);
        }
      });
      this.setState({ actionsList: treeItemsArray });
    } catch (error) {
      console.error("TOT_TOTCreateTournament_getActions \n", error);
      this.setState({
        showError: true,
        errorMessage:
          stringsConstants.TOTErrorMessage +
          " while retrieving actions list. Below are the details: \n" +
          JSON.stringify(error),
      });
    }
  }

  //Handle state values for form fields
  private handleInput(event: any, key: string) {
    switch (key) {
      case "tournamentName":
        this.setState({ tournamentName: event.target.value });
        break;
      case "tournamentDescription":
        this.setState({ tournamentDescription: event.target.value });
        break;
      default:
        break;
    }
  }

  //On select of a tree node change the state of selected actions
  private onActionSelected(items: ITreeItem[]) {
    this.setState({ selectedActionsList: items });
  }

  //Validate fields on the form and set a flag
  private ValidateFields(): Boolean {
    let validateFlag: Boolean = true;
    try {
      //clear previous error messages on the form
      this.setState({ showError: false });
      if (this.state.tournamentName == "") {
        validateFlag = false;
        this.setState({ tournamentError: true });
      } else this.setState({ tournamentError: false });
      if (this.state.selectedActionsList.length == 0) {
        validateFlag = false;
        this.setState({ actionsError: true });
      } else this.setState({ actionsError: false });
    } catch (error) {
      console.error("TOT_TOTCreateTournament_validateFields \n", error);
      this.setState({
        showError: true,
        errorMessage:
          stringsConstants.TOTErrorMessage +
          "while validating the form. Below are the details: \n" +
          JSON.stringify(error),
      });
    }
    return validateFlag;
  }

  // Save Tournament Details to SP Lists 'Tournaments' and 'Tournament Actions'
  private async saveTournament() {
    try {
      console.log(stringsConstants.TotLog + "saving tournament details.");
      let filter: string = "Title eq '" + this.state.tournamentName.trim().replace(/'/g, "''") + "'";
      if (this.ValidateFields()) {
        let commonServiceManager: commonServices = new commonServices(
          this.props.context,
          this.props.siteUrl
        );
        const allItems: any[] =
          await commonServiceManager.getItemsWithOnlyFilter(
            stringsConstants.TournamentsMasterList,
            filter
          );
        if (allItems.length == 0) {
          let submitTournamentsObject: any = {
            Title: this.state.tournamentName.trim(),
            Description: this.state.tournamentDescription,
            Status: stringsConstants.TournamentStatusNotStarted,
          };

          //Create item in 'Tournaments' list
          await commonServiceManager
            .createListItem(
              stringsConstants.TournamentsMasterList,
              submitTournamentsObject
            )
            .then((response) => {
              var selectedTreeArray: ITreeItem[] =
                this.state.selectedActionsList;
              //Loop through actions selected and create a list item for each treeview selection
              selectedTreeArray.forEach((c) => {
                //Skip parent node for treeview which is not an action
                if (c.data != undefined) {
                  let submitObject: any = {
                    Title: this.state.tournamentName.trim(),
                    Action: c.label,
                    Category: c.data.split(stringsConstants.StringSeperator)[0],
                    HelpURL: c.data.split(stringsConstants.StringSeperator)[1],
                    Points: c.subLabel
                      .split(stringsConstants.StringSeperatorPoints)[0]
                      .replace(stringsConstants.PointsReplaceString, ""),
                    Description: c.subLabel
                      .split(stringsConstants.StringSeperatorPoints)[1]
                      .replace(stringsConstants.PointsReplaceString, ""),
                  };

                  //Create an item in 'Tournament Actions' list for each selected action in tree view
                  commonServiceManager
                    .createListItem(
                      stringsConstants.TournamentActionsMasterList,
                      submitObject
                    )
                    .then((responseObj) => { })
                    .catch((error) => {
                      //Log error to console and display on the form
                      console.error(
                        "TOT_TOTCreateTournament_saveTournament \n",
                        JSON.stringify(error)
                      );
                      this.setState({
                        showError: true,
                        errorMessage:
                          stringsConstants.TOTErrorMessage +
                          "while saving the tournament action details to list. Below are the details: \n" +
                          JSON.stringify(error),
                        showForm: true,
                        showSuccess: false,
                      });
                    });
                }
              });
              this.setState({ showForm: false, showSuccess: true });
            })
            .catch((error) => {
              //Log error to console and display on the form
              console.error(
                "TOT_TOTCreateTournament_saveTournament \n",
                JSON.stringify(error)
              );
              this.setState({
                showError: true,
                errorMessage:
                  stringsConstants.TOTErrorMessage +
                  "while saving the tournament details to list. Below are the details: \n" +
                  JSON.stringify(error),
              });
            });
        } else {
          this.setState({
            showError: true,
            errorMessage:
              LocaleStrings.DuplicateTournamentNameError,
          });
        }
      }
    } catch (error) {
      console.error("TOT_TOTCreateTournament_saveTournament \n", error);
      this.setState({
        showError: true,
        errorMessage:
          stringsConstants.TOTErrorMessage +
          "while saving the tournament. Below are the details: \n" +
          JSON.stringify(error),
      });
    }
  }

  //Render Method
  public render(): React.ReactElement<ICreateTournamentProps> {
    return (
      <div className={styles.container}>
        <div className={styles.createTournamentPath}>
          <img src={require("../assets/TOTImages/BackIcon.png")}
            className={styles.backImg}
            alt={LocaleStrings.BackButton}
          />
          <span
            className={styles.backLabel}
            onClick={() => this.props.onClickCancel()}
            title={LocaleStrings.TOTBreadcrumbLabel}
          >
            {LocaleStrings.TOTBreadcrumbLabel}
          </span>
          <span className={styles.border}></span>
          <span className={styles.createTournamentLabel}>{LocaleStrings.CreateTournamentPageTitle}</span>
        </div>
        <div>
          {this.state.showSuccess && (
            <Label className={styles.successMessage}>
              <img src={require('../assets/TOTImages/tickIcon.png')} alt="tickIcon" className={styles.tickImage} />
              {LocaleStrings.CreateTournamentSuccessLabel}
            </Label>
          )}

          {this.state.showError && (
            <Label className={styles.errorMessage}>
              {this.state.errorMessage}
            </Label>
          )}
        </div>

        {this.state.showForm && (
          <div>
            <Row>
              <Col md={6}>
                <TextField
                  label={LocaleStrings.TournamentNameLabel}
                  required
                  placeholder={LocaleStrings.TournamentNameLabel}
                  maxLength={255}
                  value={this.state.tournamentName}
                  onChange={(evt) => this.handleInput(evt, "tournamentName")}
                  styles={labelStyles}
                />
                {this.state.tournamentError && (
                  <Label className={styles.errorMessage}>
                    {LocaleStrings.TournamentNameErrorLabel}
                  </Label>
                )}
              </Col>
            </Row>
            <br />
            <Row>
              <Col md={6}>
                <TextField
                  label={LocaleStrings.TournamentDescriptionLabel}
                  multiline
                  maxLength={500}
                  placeholder={LocaleStrings.TournamentDescPlaceHolderLabel}
                  value={this.state.tournamentDescription}
                  onChange={(evt) =>
                    this.handleInput(evt, "tournamentDescription")
                  }
                  styles={labelStyles}
                />
              </Col>
            </Row>
            <br />
            <Row>
              <Col className={styles.treeViewContent}>
                <div className={styles.selectTeamActionArea}>
                  <Label className={styles.selectTeamActionLabel}>
                    {LocaleStrings.SelectTeamsActionsLabel}{" "}
                    <span className={styles.asteriskStyle}>*</span>
                  </Label>
                  <TooltipHost
                    content={LocaleStrings.TeamsActionInfoToolTip}
                    calloutProps={calloutProps}
                    styles={hostStyles}
                  >
                    <Icon aria-label="Info" iconName="Info" className={classes.icon} />

                  </TooltipHost>
                </div>
                <div ref={this.createTrmtTreeViewRef}>
                  <TreeView
                    items={this.state.actionsList}
                    showCheckboxes={true}
                    selectChildrenIfParentSelected={true}
                    selectionMode={TreeViewSelectionMode.Multiple}
                    defaultExpanded={true}
                    onSelect={this.onActionSelected}
                  />
                </div>
                {this.state.actionsError && (
                  <Label className={styles.errorMessage}>
                    {LocaleStrings.ActionErrorLabel}
                  </Label>
                )}
              </Col>
            </Row>
          </div>
        )}
        <div>
          <Row>
            <Col>
              {this.state.showForm && (
                <PrimaryButton
                  text={LocaleStrings.CreateTournamentButton}
                  title={LocaleStrings.CreateTournamentButton}
                  iconProps={addIcon}
                  onClick={this.saveTournament}
                  className={styles.createBtn}
                ></PrimaryButton>
              )}
              &nbsp; &nbsp;
              <PrimaryButton
                text={LocaleStrings.BackButton}
                title={LocaleStrings.BackButton}
                iconProps={backIcon}
                onClick={() => this.props.onClickCancel()}
                styles={backBtnStyles}
              ></PrimaryButton>
            </Col>
          </Row>
        </div>
      </div> //Final DIV
    );
  }
}
