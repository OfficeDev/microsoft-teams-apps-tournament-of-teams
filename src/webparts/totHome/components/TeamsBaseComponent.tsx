import * as React from 'react';
import { render } from 'react-dom';
import { ThemeStyle, ITeamsComponentProps, ITeamsComponentState } from 'msteams-ui-components-react';

/**
 * State interface for the Teams Base user interface React component
*/
export interface ITeamsBaseComponentState extends ITeamsComponentState {
    fontSize: number;
    theme: ThemeStyle;
    badgeImgURL: string;
}

/**
 * Properties interface for the Teams Base user interface React component
*/
export interface ITeamsBaseComponentProps extends ITeamsComponentProps {
}

/**
 * Base implementation of the React based interface for the Microsoft Teams app
*/
export class TeamsBaseComponent<P extends ITeamsBaseComponentProps, S extends ITeamsBaseComponentState>
    extends React.Component<P, S> {
    constructor(props: P, state: S) {
        super(props, state);
    }

    public static render<P extends ITeamsBaseComponentProps>(element: HTMLElement, props: P) {
        render(React.createElement(this, props), element);
    }

    protected pageFontSize = () => {
        let sizeStr = window.getComputedStyle(document.getElementsByTagName('html')[0]).getPropertyValue('font-size');
        sizeStr = sizeStr.replace('px', '');
        let fontSize = parseInt(sizeStr, 10);
        if (!fontSize) {
            fontSize = 16;
        }
        return fontSize;
    }
    protected inTeams = () => {
        try {
            return window.self !== window.top;
        } catch (e) {
            return true;
        }
    }

    protected updateTheme = (themeStr: any) => {
        let theme;
        switch (themeStr) {
            case 'dark':
                theme = ThemeStyle.Dark;
                break;
            case 'contrast':
                theme = ThemeStyle.HighContrast;
                break;
            default:
                theme = ThemeStyle.Light;
        }
        this.setState({ theme });
    }

    protected getQueryVariable = (variable: any) => {
        const query = window.location.search.substring(1);
        const vars = query.split('&');
        for (const varPairs of vars) {
            const pair = varPairs.split('=');
            if (decodeURIComponent(pair[0]) === variable) {
                return decodeURIComponent(pair[1]);
            }
        }
        return null;
    }
}
