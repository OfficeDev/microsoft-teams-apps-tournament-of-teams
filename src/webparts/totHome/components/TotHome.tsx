import * as React from 'react';
import styles from '../scss/TOTHome.module.scss';
import { sp } from "@pnp/sp/presets/all";
import { Web } from "@pnp/sp/webs";
import "bootstrap/dist/css/bootstrap.min.css";
import Col from "react-bootstrap/Col";
import Media from "react-bootstrap/Media";
import Row from "react-bootstrap/Row";
import { Label, Spinner, SpinnerSize } from "office-ui-fabric-react";
import { initializeIcons } from "@uifabric/icons";
import { ITotHomeProps } from './ITotHomeProps';
import TOTLandingPage from "./TOTLandingPage";
import Header from "./Header";
import ProvisioningHelper from "../provisioning/ProvisioningHelper";
import siteconfig from "../provisioning/ProvisioningAssets.json";
import * as Constants from "../constants/strings";
import * as LocaleStrings from 'TotHomeWebPartStrings';

initializeIcons();
export interface ITotHomeState {
  showError: boolean;
  showSuccess: boolean;
  setupMessage: string;
  isShowLoader: boolean;
  enableTOT: boolean;
  isTOTEnabled: boolean;
  firstName: string;
}
//Global Variables
let rootSiteURL: string;
let spweb: any;
export default class TotHome extends React.Component<ITotHomeProps, ITotHomeState> {

  private provisioningHelper: ProvisioningHelper;

  constructor(_props: any) {
    super(_props);
    this.state = {
      showError: false,
      showSuccess: false,
      setupMessage: "",
      isShowLoader: false,
      enableTOT: false,
      isTOTEnabled: false,
      firstName: "",
    };

    let absoluteUrl = this.props.context.pageContext.web.absoluteUrl;
    let serverRelativeUrl = this.props.context.pageContext.web.serverRelativeUrl;
    if (serverRelativeUrl == "/")
      rootSiteURL = absoluteUrl;
    else
      rootSiteURL = absoluteUrl.replace(serverRelativeUrl, "");
    rootSiteURL = rootSiteURL + "/" + siteconfig.inclusionPath + "/" + siteconfig.sitename;
    sp.setup({
      spfxContext: this.props.context,
    });
    this.provisioningHelper = new ProvisioningHelper(this.props.context);
    spweb = Web(rootSiteURL);
  }


  public componentDidMount() {
    
    let givenName: any[] = this.props.context.pageContext.user.displayName.split(" ");
    if (givenName.length > 0) {
      this.setState({ firstName: givenName[0].replace(",", "") });
    }
    this.checkProvisioning();
  }

  //Check if the assets are already provisioned or not
  private async checkProvisioning() {

    this.setState({ enableTOT: false, isTOTEnabled: false });
    // Check if the assets are already provisioned or not.
    this.provisioningHelper.checkProvisioning().then((response) => {
      if (response != undefined) {
        if (!response) {
          this.setState({ enableTOT: true });
          console.log(Constants.ProvisioningLog, "enableTOT ", this.state.enableTOT);
        }
        else {
          this.setState({ isTOTEnabled: true });
          console.log(Constants.ProvisioningLog, "isTOTEnabled ", this.state.isTOTEnabled);
        }
      }
    });
  }

 //This function is called when Enable TOT image is clicked, to create a site and all required assets  
  public enableAppSetup = () => {
    this.setState({ isShowLoader: true, setupMessage: LocaleStrings.ProvisioningSetupMessage, enableTOT: false });
    //Creating provisioning assets for the App
    this.provisioningHelper.createSiteAndLists().then((response) => {
      if (response != undefined) {
        if (!response) {
          this.setState({ showError: true, showSuccess: false, enableTOT: true, isShowLoader: false, setupMessage: LocaleStrings.ProvisioningErrorMessage });
          console.log(Constants.ProvisioningLog, "Error in Provisioning. ");
        }
        else {
          this.setState({ showError: false, showSuccess: true, enableTOT: false, isShowLoader: false, setupMessage: LocaleStrings.ProvisioningSuccessMessage });
          console.log(Constants.ProvisioningLog, "Provisioning Successful. ");
        }
      }
    }).catch((err) => {
      this.setState({ showError: true, showSuccess: false, enableTOT: true, isShowLoader: false, setupMessage: LocaleStrings.ProvisioningErrorMessage  });
      console.error("PVSS_Home_enableAppSetup. \n ", err);
    });
  }


  public render(): React.ReactElement<ITotHomeProps> {
    return (
      <div className={styles.totHome} >
        <div className={styles.container} >
          <div>
            <Header
              clickcallback={() =>
                this.checkProvisioning()
              }
            />
          </div>
          {this.state.enableTOT && !this.state.showSuccess && (
            <div className={styles.imgheader}>
              <span className={styles.totPageHeading}>{LocaleStrings.WelcomeLabel} {this.state.firstName}!</span>
            </div>
          )}
          <div>
            {
              this.state.isTOTEnabled && (
                <TOTLandingPage
                  siteUrl={this.props.siteUrl}
                  context={this.props.context}
                  isTOTEnabled={this.state.isTOTEnabled}
                  firstName={this.state.firstName}
                />
              )
            }
          </div>
          <div className={styles.background}>
            {this.state.isShowLoader && (
              <Label className={styles.setupMessage}>
                {this.state.setupMessage}
              </Label>
            )}
            {this.state.isShowLoader && (
              <Spinner
                label={LocaleStrings.SpinnerMessage}
                size={SpinnerSize.large}
              />
            )}
            {this.state.showError && (
              <Label className={styles.errorMessage}>
                {this.state.setupMessage}
              </Label>
            )}
            {this.state.showSuccess && (
              <Label className={styles.successMessage}>
                {this.state.setupMessage}
              </Label>
            )}
            {this.state.enableTOT && !this.state.showSuccess && (
              <div>
                <div className={styles.grid}>
                  <Row className="mt-4">
                    <Col sm={3} className={styles.imageLayout}>
                      <Media
                        className={styles.cursor}
                        onClick={() => this.enableAppSetup()}
                      >
                        <div className={styles.mb}>
                          <img
                            src={require("../assets/TOTImages/EnableTOT.svg")}
                            alt={LocaleStrings.EnableTOTToolTip}
                            title={LocaleStrings.EnableTOTToolTip}
                            className={styles.dashboardimgs}
                          />
                          {!this.state.isTOTEnabled && (<div className={`${styles.center} ${styles.enableTournamentLabel}`} title={LocaleStrings.EnableTOTLabel}>
                            {LocaleStrings.EnableTOTLabel}</div>)}
                        </div>
                      </Media>
                    </Col>
                  </Row>
                </div>
              </div>
            )}
          </div>
        </div >
      </div >
    );
  }
}
