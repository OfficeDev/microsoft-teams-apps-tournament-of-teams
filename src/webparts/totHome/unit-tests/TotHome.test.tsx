import { WebPartContext } from '@microsoft/sp-webpart-base';
import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import { SPWebPartContextMock } from "spfx-ut-library/lib/base/SPWebPartContextMock";
import { JestHelper } from "spfx-ut-library/lib/helpers/JestHelper";
import { ITotHomeProps } from '../components/ITotHomeProps';
import TotHome from '../components/TotHome';

JestHelper.registerMocks(jest);

jest.mock('@pnp/spfx-controls-react/lib/TreeView', () => 'TreeView');
jest.mock('react-bootstrap/esm/Card', () => 'Card');
jest.mock('react-bootstrap/esm/Col', () => 'Col');
jest.mock('react-bootstrap/esm/Row', () => 'Row');
jest.mock('@pnp/spfx-controls-react/lib/ChartControl', () => 'ChartControl');


describe('TOT Home Component', () => {
    let wrapper: ReactWrapper<ITotHomeProps, {}>;

    let mockCtx = new SPWebPartContextMock();
    const context = mockCtx as unknown as WebPartContext;
    const siteUrl = "https://siteUrl.com";
    const description = "this is a description";

    beforeEach(() => {
        wrapper = mount(React.createElement(
            TotHome, {
            siteUrl: siteUrl,
            context: context,
            description: description
        }
        ));
    });
    afterEach(() => {
        wrapper.unmount();
    });

    test("Check if the component got rendered", () => {
        expect(wrapper.exists()).toBe(true);
    });

    test("Check if onclick of Home Logo calls the home call back function", () => {
        const homeLogo = wrapper.find('.clbHeaderLogo');
        homeLogo.simulate('click');
        expect(wrapper.state('enableTOT')).toEqual(false);
    });

    test("Check if onclick enableTot Image enables the Tournament of Teams", async () => {
        wrapper.setState({
            enableTOT: true,
            showSuccess: false
        });
        const enableTOTLabel = wrapper.find('.enableTournamentLabel');
        enableTOTLabel.simulate('click');
        expect(wrapper.state('enableTOT')).toEqual(false);
    });

    test('Check if the component matches the snapshot', () => {
        expect(wrapper.html).toMatchSnapshot();
    });
});