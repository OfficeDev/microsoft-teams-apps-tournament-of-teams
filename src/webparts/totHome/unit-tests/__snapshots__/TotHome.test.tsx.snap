// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TOT Home Component Check if the component matches the snapshot 1`] = `[Function]`;
